src/components/SideBar/SidebarTags.tsx

```diff
@@ -46,6 +45,19 @@ export const SidebarTags: FunctionComponent<SidebarTagsProps> = ({tags, selected
         } as Record<DisplayedTagsType, Array<APILabel>>
     ), [tags]);
 
+    const updateSelectedTags = (selectedTags: ReadonlyArray<string>, tagId: string, isChecked: boolean) => {
+        const index = selectedTags.indexOf(tagId);
+        const newSelectedTags = [...selectedTags]
+
+        if (index === -1 && isChecked) {
+            newSelectedTags.push(tagId);
+        } else if (index !== -1 && !isChecked) {
+            newSelectedTags.splice(index, 1);
+        }
+
+        return newSelectedTags
+    }
+
```

Comment:
I'm unsure of this one, the previous version uses the previous state and this seems to be using a stale state. I think we need to do something at zustand level to grant access to the previous state like the `useState` does. `immer` (with the `produce` function) helps to manage the immutability to prevent issues when merging old and new state.

Reply:
There is a [.getState() in zustand](https://github.com/pmndrs/zustand#readingwriting-state-and-reacting-to-changes-outside-of-components) that gives you a fresh state without any reaction, like `useStore.getState().name`. But I decided to introduce a new action in the config store that uses `immer` `produce` to update single selected tags. Lmk, what you think!